saleor/core/permissions.py

```diff
@@ -0,0 +1,45 @@
+from django.contrib.auth.models import Permission
+from django.shortcuts import get_object_or_404
+
+from saleor.userprofile.models import User
+
+
+MODELS_PERMISSIONS = (('view', 'View Products in Dashboard'),
+                      ('edit', 'Edit Product in Dashboard'))
+
+PERMISSIONS = set([permission[0] for permission in MODELS_PERMISSIONS])
```

Comment:
Names `MODELS_PERMISSIONS` and `PERMISSIONS` are a bit confusing when you first read this, because it's hard to guess what is the purpose of having the two constants with similar names and the second being just the transformation of the first one. `PERMISSIONS` is used only once in `update_permissions` so maybe it should be a local variable. And if `MODELS_PERMISSIONS` is meant to be imported across the project models then this  could be the only constant.

Reply:
Yea i know this looks odd. I will change it to local variable.

Reply:
By the way, set comprehension has been a thing for at least five years 😄